web/src/lib/services/api.ts

```diff
@@ -44,6 +44,11 @@ function fetchAPI(path, method, tokenValue, json, checkResponse) {
                 if (tokenValue) {
                     console.log("Error 401: Unauthorized. Deleting the token.");
                     logout(false);
+
+                    loginModalState.set({
+                        openRequested: true,
```

Comment:
now i am thinking, instead of always setting this store value all over the place, wouldn't it be nicer to have a function in utils.ts:

`function requestLoginModal(callbackFunc: () => void = () => {}) { loginModalState.set({openRequested: true, callbackFunc}) }`

then we can just call `requestLoginModal()` here, or pass the optional callback if needed...